keps/sig-api-machinery/2896-openapi-v3/README.md

```diff
@@ -547,28 +585,40 @@ It's impossible to provide comprehensive guidance, but at the very high level
 These goals will help you determine what you need to measure (SLIs) in the next
 question. -->
 
+This feature should not affect the SLO of any components.
+
+The size of the OpenAPI scales as a cluster has more object types. In OpenAPI V2, aggregating the OpenAPI for all types was very resource intensive. In OpenAPI V3, aggregation is split and the aggregator will act as a proxy instead. We will monitor the current metrics to ensure that no regression is introduced with OpenAPI V3.
+
 ###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
 
 <!-- Pick one more of these and delete the rest. -->
 
-- [ ] Metrics
-  - Metric name:
-  - [Optional] Aggregation method:
-  - Components exposing the metric:
-- [ ] Other (treat as last resort)
-  - Details:
+A new metric will be added in the aggregator to measure response time for OpenAPI specific requests.
+
+ - [X] Metrics
+  - Metric name: `apiserver_openapi_request_duration_seconds`
```

Comment:
where does CRD aggregation of the spec happen? Is that in the handler code path or asynchronous via controller? In the later case, a metric for that computation makes more sense. Serving some blob from a cache is cheap. Not worth a metric.

Reply:
The CRD aggregation is asynchronous via a controller. It wouldn't hurt to have a metric for that computation, I'll change this section